mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -129,6 +128,8 @@
       scan = scan.filter(filter);
     }
 
+    FileIO io = table.io();
```

Comment:
nit: any reason for saving them into variables?

Reply:
Yeah.
We need to use them in the lambda. Can not use the `table` because it is not `finallike`. So either I have to do some magic like:
```
    Table nonFinalTable = HiveIcebergStorageHandler.table(conf, conf.get(InputFormatConfig.TABLE_IDENTIFIER));
    Table table = nonFinalTable != null ? nonFinalTable : Catalogs.loadTable(conf);
```
or I have do the same with the `io`, `encryption`.

I call for a vote which is uglier 😄

Reply:
I see. Would this make it effectively final?
```
Table table = Optional.ofNullable(HiveIcebergStorageHandler.table(conf, conf.get(InputFormatConfig.TABLE_IDENTIFIER)))
    .orElseGet(() -> Catalogs.loadTable(conf));
```

Reply:
That's much better. Done.
Thanks